rfd/0081/health-check-draft.md

```diff
@@ -0,0 +1,58 @@
+# Mariposa Health Check Service
+
+The Mariposa (RFD36) health check service will be responsible for monitoring the health of user services. It establishes a set of supported health checks and provides an API for defining health checks that should be ran against project services. These health check definitions will be persisted in a proper data store (Postgres/Manatee?).
```

Comment:
I'd propose that the Healthcheck service runs in a [sdc-nat zone](https://github.com/joyent/sdc-nat) associated with the project. The NAT zone is something we operate as a service, but the customer doesn't have access to, and it's connected to the private networks of the user.

There are some cases we'll have to discuss:

- What if the NAT zone fails?
- What if the user has multiple fabric networks within a project?
- What if the user has instances in a project that are not connected to any NAT?